### ISSUE TITLE
vscode extension not running - how to debug?

### ISSUE THREAD

**Issue by brahn** [microsoft/pyright#648]
**Describe the bug**
Despite having installed the pyright VSCode extension I am not getting any feedback from vscode about my type errors.

I expect that there is some basic setup that I'm misunderstanding, so perhaps the "bug" is some missing piece of documentation or feedback, or just me overlooking something.  Thanks in advance for any pointers!

**To Reproduce**

* Running VSCode 1.44.2 on MacOS Catalina 10.15.4
* Installed VSCode extensions: `ms-python.python` (2020.4.76186), `ms-pyright.pyright` (1.1.34)
* Python runtime: Python 3.7.7 64-bit

 Created a `scratchpad.py` file with the following contents:
```py
from typing import String

x = '1'
y = 2
z = x + y  # should be a type error (this isn't javascript!)
w=x+y  # missing whitespace


def foo(x: String) -> String:
    return 0  # returning the wrong type

foo(1)  # calling with the wrong type
```

My screen now looks like this -- I see the whitespace error reported by flake8, but no pyright-reported errors either in syntax highlighting or the "problems" window:

![scratchpad py 2020-05-01 12-19-14](https://user-images.githubusercontent.com/235302/80834737-246a4e00-8ba6-11ea-8341-2ca790f037da.png)

**Expected behavior**
Highlighted type errors

**Comment by erictraut** [comment]
You shouldn't need to do any special setup. It should just work. Let's try to diagnose the problem.

Please open the "Output" tab and choose "Pyright" from the combo box. This will display diagnostic information. Are there any clues there? Perhaps a mention of a crash or exception?

**Comment by brahn** [comment]
Ah, I think that was exactly the pointer I needed!  Here was the output:

```
Pyright language server starting
Server root directory: /Users/brahn/.vscode/extensions/ms-pyright.pyright-1.1.34
Starting service instance "data-infra"
Background analysis root directory: /Users/brahn/.vscode/extensions/ms-pyright.pyright-1.1.34
Background analysis started
Received updated settings
Loading configuration file at /Users/brahn/pyrightconfig.json
```
And then it just hangs.  

When I removed the file at `/Users/brahn/pyrightconfig.json` (which I had forgotten about in my original writeup since that was part of my throw-everything-at-the-problem approach) instead I see some reasonable output and the type errors are highlighted as expected!

For reference, the problematic pyrightconfig.json file (which I had set up for a different project) had these contents:

```json
{
    "executionEnvironments": [
        {
            "root": "charts"
        }
    ]
}
```

Thanks again!  Feel free to close as resolved or let me know if more info would be useful.

**Comment by erictraut** [comment]
I'd like to determine the cause of the hang. I've tried to repro the situation you've reported, but I'm not seeing any hang. 

A few follow-up questions for you...
1. Do you have a directory called "charts" within your workspace? If so, what's in it? Is it empty, does it contain thousands of python source files?
2. When you see the hang, is "Loading configuration file at /Users/brahn/pyrightconfig.json" the last thing you see in the output? I would expect, at a minimum, that you would see another line in the output that says "No include entries specified; assuming XXX".
3. Is it possible that your pyrightconfig.json file had some corrupt character or syntax that's not represented in the JSON snippet you posted above?

Thanks for your help in tracking this down!

**Comment by erictraut** [comment]
I'm not able to repro the problem, so I'm going to close it for now. If you have additional information, please add details to the bug, and I'll investigate further.

**Comment by brahn** [comment]
Hi @erictraut my apologies for the slow reply, especially since you were so prompt and helpful with my original question!  Here's what I've found --

I can definitely reproduce the problem by adding back the file at `~/pyrightconfig.json`.  However, restarting the server (via the `Pyright: restart server` vscode command) was not sufficient to repro, I had to quite vscode and restart by typing `code .` inside the project subdirectory (which is a couple of subdirectory levels deep  in `~`).

> 1. Do you have a directory called "charts" within your workspace? If so, what's in it? Is it empty, does it contain thousands of python source files?

It is non-empty, but it's pretty small (three modest `.py` files). 

> 2. When you see the hang, is "Loading configuration file at /Users/brahn/pyrightconfig.json" the last thing you see in the output? I would expect, at a minimum, that you would see another line in the output that says "No include entries specified; assuming XXX".

Yes, that's the last thing I see:

<img width="766" alt="pyright-output" src="https://user-images.githubusercontent.com/235302/81335072-e1075800-905b-11ea-8e2b-c6727b5acd23.png">


> 3. Is it possible that your pyrightconfig.json file had some corrupt character or syntax that's not represented in the JSON snippet you posted above?

Interesting hypothesis!  But no, I reproduce the error by copying the snippet above and pasting into a new `pyright.config file

Thanks again!  Please let me know if there's more I can do to help investigate.
